src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -66,7 +66,7 @@
     public static final String EXCEPTION_MSG = "general.exception";
 
     /** Logger for Checker. */
-    private static final Log LOG = LogFactory.getLog(Checker.class);
+    private final Log log;
```

Comment:
ok, please focus on my point "2)"  to enforce this by Checkstyle rule , Main.java should be added to suppression https://github.com/checkstyle/checkstyle/blob/master/config/sevntu_suppressions.xml